### ISSUE TITLE
Crash during x.509 certificate inspection (net::CertVerifyProcMac::VerifyInternal)

### ISSUE THREAD

**Issue by CharlieHess** [electron/electron#11408]
* Electron version: 1.7.x / 1.8.x
* Operating system: macOS 10.12.6

We're seeing a crash during an SSO login, on both `1.7.x` and `1.8.x`.

<details>

```
Process: Slack [36927] 
Path: /Applications/Slack.app/Contents/MacOS/Slack 
Identifier: com.tinyspeck.slackmacgap 
Version: 2.9.0 (4332) 
Code Type: X86-64 (Native) 
Parent Process: ??? [1] 
Responsible: Slack [36927] 
User ID: 1486302471

Date/Time: 2017-12-08 19:02:22.663 -0500 
OS Version: Mac OS X 10.12.6 (16G1114) 
Report Version: 12 
Anonymous UUID: 4AF90C1D-44B6-0CC0-FA2D-A54077113AD2

Sleep/Wake UUID: 55E07179-6D14-4D76-9BC8-F3B75E3FAEDB

Time Awake Since Boot: 11000 seconds 
Time Since Wake: 42 seconds

System Integrity Protection: disabled

Crashed Thread: 3 WorkerPool/14351

Exception Type: EXC_BAD_ACCESS (SIGSEGV) 
Exception Codes: KERN_INVALID_ADDRESS at 0xfffffffffffffff8

VM Regions Near 0xfffffffffffffff8: 
--> shared memory 00007ffffffe8000-00007ffffffe9000 [ 4K] r-x/r-x SM=SHM

Thread 0:: CrBrowserMain Dispatch queue: com.apple.main-thread 
0 libsystem_kernel.dylib 0x00007fffbbd8334a mach_msg_trap + 10 
1 libsystem_kernel.dylib 0x00007fffbbd82797 mach_msg + 55 
2 com.apple.CoreFoundation 0x00007fffa64d7434 __CFRunLoopServiceMachPort + 212 
3 com.apple.CoreFoundation 0x00007fffa64d68c1 __CFRunLoopRun + 1361 
4 com.apple.CoreFoundation 0x00007fffa64d6114 CFRunLoopRunSpecific + 420 
5 com.apple.HIToolbox 0x00007fffa5a36ebc RunCurrentEventLoopInMode + 240 
6 com.apple.HIToolbox 0x00007fffa5a36cf1 ReceiveNextEventCommon + 432 
7 com.apple.HIToolbox 0x00007fffa5a36b26 _BlockUntilNextEventMatchingListInModeWithFilter + 71 
8 com.apple.AppKit 0x00007fffa3fcda54 _DPSNextEvent + 1120 
9 com.apple.AppKit 0x00007fffa47497ee -[NSApplication(NSEvent) _nextEventMatchingEventMask:untilDate:inMode:dequeue:] + 2796 
10 com.apple.AppKit 0x00007fffa3fc23db -[NSApplication run] + 926 
11 com.github.electron.framework 0x000000010a3900de 0x10a158000 + 2326750 
12 com.github.electron.framework 0x000000010a38f74c 0x10a158000 + 2324300 
13 com.github.electron.framework 0x000000010a38bc8e 0x10a158000 + 2309262 
14 com.github.electron.framework 0x000000010a3a1653 0x10a158000 + 2397779 
15 com.github.electron.framework 0x000000010a5f187b 0x10a158000 + 4823163 
16 com.github.electron.framework 0x000000010a5f16d0 0x10a158000 + 4822736 
17 com.github.electron.framework 0x000000010a5f3e12 0x10a158000 + 4832786 
18 com.github.electron.framework 0x000000010a5ed7ac 0x10a158000 + 4806572 
19 com.github.electron.framework 0x000000010a55330d 0x10a158000 + 4174605 
20 com.github.electron.framework 0x000000010a552406 0x10a158000 + 4170758 
21 com.github.electron.framework 0x000000010a159d07 AtomMain + 71 
22 com.tinyspeck.slackmacgap 0x000000010a152f26 main + 38 
23 libdyld.dylib 0x00007fffbbc5c235 start + 1

Thread 1: 
0 libsystem_kernel.dylib 0x00007fffbbd8b44e __workq_kernreturn + 10 
1 libsystem_pthread.dylib 0x00007fffbbe7548e _pthread_wqthread + 1023 
2 libsystem_pthread.dylib 0x00007fffbbe7507d start_wqthread + 13

Thread 2: 
0 libsystem_kernel.dylib 0x00007fffbbd8b44e __workq_kernreturn + 10 
1 libsystem_pthread.dylib 0x00007fffbbe7548e _pthread_wqthread + 1023 
2 libsystem_pthread.dylib 0x00007fffbbe7507d start_wqthread + 13

Thread 3 Crashed:: WorkerPool/14351 
0 com.github.electron.framework 0x000000010b97d4bb 0x10a158000 + 25318587 
1 com.github.electron.framework 0x000000010b97c275 0x10a158000 + 25313909 
2 com.github.electron.framework 0x000000010ba473f2 0x10a158000 + 26145778 
3 com.github.electron.framework 0x000000010ba47f78 0x10a158000 + 26148728 
4 com.github.electron.framework 0x000000010a397208 0x10a158000 + 2355720 
5 com.github.electron.framework 0x000000010a3e3061 0x10a158000 + 2666593 
6 com.github.electron.framework 0x000000010a396877 0x10a158000 + 2353271 
7 libsystem_pthread.dylib 0x00007fffbbe7593b _pthread_body + 180 
8 libsystem_pthread.dylib 0x00007fffbbe75887 _pthread_start + 286 
9 libsystem_pthread.dylib 0x00007fffbbe7508d thread_start + 13
```

</details>

--------------

We ran `electron-atos` on the crashed thread and the result was:

```
net::CertVerifyProcMac::VerifyInternal(net::X509Certificate*, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int, net::CRLSet*, std::__1::vector<scoped_refptr<net::X509Certificate>, std::__1::allocator<scoped_refptr<net::X509Certificate> > > const&, net::CertVerifyResult*) (in Electron Framework) (cert_verify_proc_mac.cc:352)
net::CertVerifyProc::Verify(net::X509Certificate*, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int, net::CRLSet*, std::__1::vector<scoped_refptr<net::X509Certificate>, std::__1::allocator<scoped_refptr<net::X509Certificate> > > const&, net::CertVerifyResult*) (in Electron Framework) (cert_verify_proc.cc:420)
net::DoVerifyOnWorkerThread(scoped_refptr<net::CertVerifyProc> const&, scoped_refptr<net::X509Certificate> const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int, scoped_refptr<net::CRLSet> const&, std::__1::vector<scoped_refptr<net::X509Certificate>, std::__1::allocator<scoped_refptr<net::X509Certificate> > > const&, int*, net::CertVerifyResult*) (in Electron Framework) (multi_threaded_cert_verifier.cc:193)
base::internal::Invoker<base::internal::BindState<void (*)(scoped_refptr<net::CertVerifyProc> const&, scoped_refptr<net::X509Certificate> const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int, scoped_refptr<net::CRLSet> const&, std::__1::vector<scoped_refptr<net::X509Certificate>, std::__1::allocator<scoped_refptr<net::X509Certificate> > > const&, int*, net::CertVerifyResult*), scoped_refptr<net::CertVerifyProc>, scoped_refptr<net::X509Certificate>, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, int, scoped_refptr<net::CRLSet>, std::__1::vector<scoped_refptr<net::X509Certificate>, std::__1::allocator<scoped_refptr<net::X509Certificate> > >, int*, net::CertVerifyResult*>, void ()>::Run(base::internal::BindStateBase*) (in Electron Framework) (bind_internal.h:339)
base::(anonymous namespace)::PostTaskAndReplyRelay::RunTaskAndPostReply() (in Electron Framework) (post_task_and_reply_impl.cc:47)
base::(anonymous namespace)::WorkerThread::ThreadMain() (in Electron Framework) (worker_pool_posix.cc:93)
base::(anonymous namespace)::ThreadFunc(void*) (in Electron Framework) (platform_thread_posix.cc:73)
```

### How to reproduce
We've been unable to reproduce; I'll update the bug as we get more information.

**Comment by sofianguy** [comment]
@CharlieHess I'm going to close this to lessen the issue tracker. Feel free to reopen if you're able to reproduce the issue.

**Comment by CharlieHess** [comment]
While we can't reproduce it internally at Slack, we're still getting periodic reports from customers, even on the latest Electron version. Going to reopen this since it seems legit.

**Comment by sofianguy** [comment]
@CharlieHess Are you still experiencing this crash on a more recent version of Electron?

**Comment by sofianguy** [comment]
@CharlieHess We are no longer releasing bug fixes for versions earlier than 2.0.x. Feel free to reopen if you see this bug again in versions later than 2.0.x
